dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/CompanyOwnershipChecker.kt

```diff
@@ -56,4 +58,44 @@ class CompanyOwnershipChecker(
         val companyId = dataMetaInformationManager.getDataMetaInformationByDataId(dataId).company.companyId
         return isCurrentUserCompanyOwnerForCompany(companyId)
     }
+
+    /**
+     * Method to check whether a given company has at least one company owner
+     * @param companyId the ID of the company
+     * @return a Boolean indicating whether the company has at least one company owner
+     */
+    @Transactional(readOnly = true)
+    fun companyExistsAndHasNoOwner(companyId: String): Boolean {
+        val companyOwners = companyRolesControllerApi.getCompanyRoleAssignments(
+            CompanyRole.CompanyOwner, UUID.fromString(companyId),
+        )
+        return companyOwners.isEmpty()
+    }
+
+    // This function can be made more generic if additional field-specific checks are needed in the future
+    /**
+     * Method to check whether the patch contains only fields that are allowed to be altered by the uploader
+     * @param patch the fields to be patched
+     * @return a Boolean indicating whether the patch complies with the access requirements
+     */
+    @Transactional(readOnly = true)
+    fun onlyPatchesAuthorizedFields(patch: CompanyInformationPatch): Boolean {
```

Comment:
Funktionsnamen sollten am besten immer ein Verb beinhalten.
z.B.  `areOnlyAuthorizedFieldsPatched()`

Da aktuell aber sowieso nur gecheckt werden muss ob contactDetails gepatcht werden, kannst du hier auch schreiben:

`isOnlyContactDetailsBeingPatched()`

Reply:
both company contatc details as well as website

renamed

Reply:
Achso ok. Also sollen company-owner auch "website" patchen dürfen?
Davon steht nichts direkt in der description des Tickets: https://jira.d-fine.dev/browse/DALA-4303

Bist du sicher?

Reply:
ja ich habe nochmal mit Andreas darüber gesprochen

Reply:
Alles klar. Dann passt das